sklearn/linear_model/tests/test_ridge.py

```diff
@@ -837,6 +839,70 @@ def test_ridge_fit_intercept_sparse():
         assert_array_almost_equal(dense.coef_, sparse.coef_)
 
 
+@pytest.mark.parametrize('return_intercept', [False, True])
+@pytest.mark.parametrize('sample_weight', [None, np.ones(1000)])
+@pytest.mark.parametrize('arr_type', [np.array, sp.csr_matrix])
+@pytest.mark.parametrize('solver', ['auto', 'sparse_cg', 'cholesky', 'lsqr',
+                                    'sag', 'saga'])
+def test_ridge_regression_check_arguments_validity(return_intercept,
+                                                   sample_weight, arr_type,
+                                                   solver):
+    """check if all combinations of arguments give valid estimations"""
+
+    # test excludes 'svd' solver because it raises exception for sparse inputs
+
+    rng = check_random_state(42)
+    X = rng.rand(1000, 3)
+    true_coefs = [1, 2, 0.1]
+    y = np.dot(X, true_coefs)
+    intercept = 0.
+    if return_intercept:
+        intercept = 10000.
+    y += intercept
+    X_testing = arr_type(X)
+
+    alpha, atol, tol = 1e-3, 1e-4, 1e-6
+    target = ridge_regression(X_testing, y, alpha=alpha,
+                              solver=solver,
+                              sample_weight=sample_weight,
+                              return_intercept=return_intercept,
+                              tol=tol,
+                              )
+
+    if return_intercept:
+        coef, intercept = target
+        assert_allclose(coef, true_coefs, rtol=0, atol=atol)
+        assert_allclose(intercept, intercept, rtol=0, atol=atol)
```

Comment:
Please change the name of the true intercept to `true_intercept` because this doesn't check anything

Reply:
True! Thanks!